### DISCUSSION TITLE
useStorage: is key being static by design?

### DISCUSSION THREAD

**Discussion by PindaPixel** [vueuse/vueuse#3724] [Q&A] [answered]
I noticed the storage key in useStorage is not a `MaybeRefOrGetter` but rather a string. Is this by design? 

My use case for it being reactive is I have an instance where the key depends on a dynamic route parameter (e.g., `/resource/:id`), and switching between pages does not unmount the component. I could add a key to the page to remount it on navigation, but to me it would make sense to make the key a getter or computed value.

**Comment by ilyaliao** [comment, marked as answer]
Implemented through #4464!
